framework/physics/FieldFunction/ff_gridbased_00_constrdestr.cc

```diff
@@ -134,12 +137,25 @@ chi_math::SDMPtr FieldFunctionGridBased::MakeSpatialDiscretization(
   typedef chi_math::spatial_discretization::FiniteVolume FV;
   typedef chi_math::spatial_discretization::PieceWiseLinearContinuous PWLC;
   typedef chi_math::spatial_discretization::PieceWiseLinearDiscontinuous PWLD;
+  typedef chi_math::spatial_discretization::LagrangeContinuous LagC;
+  typedef chi_math::spatial_discretization::LagrangeDiscontinuous LagD;
 
-  if (sdm_type == "FiniteVolume") return FV::New(*grid_ptr);
+  if (sdm_type == "FV") return FV::New(*grid_ptr);
 
-  chi_math::QuadratureOrder q_order = chi_math::QuadratureOrder::SECOND;
-  chi_math::CoordinateSystemType coordinate_system_type =
+  chi_math::CoordinateSystemType cs_type =
     chi_math::CoordinateSystemType::CARTESIAN;
+  std::string cs = "cartesian";
+  if (user_params.Has("coordinate_system"))
+  {
+    cs = params.GetParamValue<std::string>("coordinate_system");
+
+    using namespace chi_math;
+    if (cs == "cartesian") cs_type = CoordinateSystemType::CARTESIAN;
+    if (cs == "rz") cs_type = CoordinateSystemType::CYLINDRICAL;
+    if (cs == "1d_spherical") cs_type = CoordinateSystemType::SPHERICAL;
```

Comment:
It feels icky to me to have coordinate system and dimensionality inherently tied together in the input language for cylindrical and spherical, but not cartesian. It would be could to see `cartesian`, `cylindrical`, and  `spherical` instead with subsequent checks on the mesh dimension for validity.

Reply:
IDK. I've had the same icky feeling when seeing cartesian (X, XY, XYZ), cylindrical (R, RZ, RZ-Varphi), spherical (R, R-Varphi, R-Varphi-Theta) in other codes. There is always confusion on the dimensionality when one does that. There are few examples of codes that do the 3D cylindrical and 2D+3D spherical variant.  In fact, Kernels, and stuff become substantially more complicated with the cylindrical 3D variants. I particularly put this in here to concisely remind the user that this is the intended dimension. This is what moose has:
```
MultiMooseEnum coord_types("XYZ RZ RSPHERICAL", "XYZ");
```

I did identify now that we can by accident try to make an rz field with 1D cells. Created issue #396 for this.

Reply:
Maybe you misunderstood my comment. These inputs specify the coordinate system type. Cartesian, cylindrical, and spherical geometries can be 1d, 2d, or 3d. 

I am *not* recommending to use `x`, `xy`, and `xyz`-type options, but rather, dimension independent `cartesian`, `cylindrical`, and `spherical` options. For whatever the user is trying to do, if a particular coordinate system-dimensionality pair is not implemented, the code should puke. 

Why not name the options after the options in the `CoordinateSystemType` enum class? If we had full 3D in all coordinate systems, then if a user specifies `1d_cylindrical`, `rz`, or `3d_cylindrical`, it would result in `cs_type = CoordinateSystemType::CYLINDRICAL`, which makes specifying the coordinate system in a dimensionality-dependent way unnecessary and makes the input language less clear (especially when considering 1d, 2d, and 3d Cartesian is just `cartesian`).